docs/language-server/protocol-language-server.md

```diff
@@ -1507,6 +1552,111 @@ table FileContentsReply {
   access to a resource.
 - [`FileNotFound`](#filenotfound) informs that file cannot be found.
 
+### `file/writeBytes`
+
+This requests that the file manager component writes a set of bytes to the
+specified file at the specified offset.
+
+- **Type:** Request
+- **Direction:** Client -> Server
+- **Connection:** Binary
+- **Visibility:** Public
+
+This method will create a file if no file is present at `path`.
+
+#### Parameters
+
+```idl
+namespace org.enso.languageserver.protocol.binary;
+
+table WriteBytesRequest {
+  // The file to write to.
+  path : Path (required);
+
+  // The byte offset in the file to write from.
+  byteOffset : ulong (required);
```

Comment:
A few things should be specified. What offset value should be set when appending to the file, equal to the file length or length + 1? What should happen when the offset is greater than the file length (there is a gap)? For example, [Java API](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/nio/channels/SeekableByteChannel.html#position(long)) would fill the gap with junk bytes, which is probably not desirable.